maestro/src/agents/openai_agent.py

```diff
@@ -1,87 +1,394 @@
+                        elif event.name == "tool_output":
+                            tool_output = getattr(event.item, 'output', 'N/A')
+                            self.print(f"DEBUG [OpenAIAgent {self.agent_name}]: Finished tool call. Output: {str(tool_output)[:100]}...")
+                        elif event.name == "message_output_created":
+                            # message_text = ItemHelpers.text_message_output(event.item) # Can be verbose
+                            self.print(f"DEBUG [OpenAIAgent {self.agent_name}]: Message output item created.")
+                            pass
+                        elif event.name == "run_completed":
+                            self.print(f"DEBUG [OpenAIAgent {self.agent_name}]: Agent stream processing finished (run_item_stream_event: {event.name}).")
+                        else:
+                            self.print(f"DEBUG [OpenAIAgent {self.agent_name}]: Received run item event: {event.name}")
+
+                    elif event.type == "agent_updated_stream_event":
+                        if last_event_was_delta:
+                            print("")
+                            last_event_was_delta = False
+                        self.print(f"DEBUG [OpenAIAgent {self.agent_name}]: Agent updated to: {event.new_agent.name}")
+                    else:
+                        if last_event_was_delta:
+                            print("")
+                            last_event_was_delta = False
+                        self.print(f"DEBUG [OpenAIAgent {self.agent_name}]: Received unknown event type: {event.type}")
+
+                if last_event_was_delta:
+                    print("")
+
+        except Exception as e:
+            if last_event_was_delta:
+                print("")
+            error_msg = f"ERROR [OpenAIAgent {self.agent_name}]: Agent stream failed: {e}"
+            self.print(error_msg)
+            self.print(traceback.format_exc())
+            return f"Error during agent streaming execution: {e}"
+
+        # Create the final output from all the bits we've received
+        final_output_str = "".join(final_output_chunks)
+
+        self.print(f"Final Response from {self.agent_name} (streaming collected): {final_output_str}")
+
+        return final_output_str
+
     async def run(self, prompt: str) -> str:
         """
-        Runs the agent with the given prompt.
+        Runs the agent with the given prompt, potentially overriding to streaming
+        based on MAESTRO_OPENAI_STREAMING environment variable.
+
         Args:
             prompt (str): The prompt to run the agent with.
         """
-        self.print(f"Running {self.agent_name}...")
-        result = await OAIRunner.run(self.openai_agent, prompt)        
-        self.print(f"Response from {self.agent_name}: {result.final_output}")
-        return result.final_output
+        streaming_override = os.getenv("MAESTRO_OPENAI_STREAMING", "auto").lower()
```

Comment:
It seems that this `run` and `run_streaming` can be simplified by using the common code.